text/XXXX-delete_and_drop.md

```diff
@@ -0,0 +1,225 @@
+
+```
+$ qri drop me/dataset
+error: cannot drop a dataset you own
+
+because you are the author of this dataset, dropping it would remove the 
+authoritative history of this dataset. Instead of dropping, you can 
+delete this dataset:
+
+  $ qri delete --all me/dataset
+
+```
+
+Drop should do the opposite of a `pull` operation, removing from logbook, caches, and stroed versions. This solves a business-logic problem: anything we do "automatically" for a user on pull needs to be undone when the user wants to go the other way. Having an explicit "opposite of pull" command allows us to write the converse business logic to whatever pull does automatically, restoring the user to a clean state without having to resort to plumbing commands.
+
+From the UX side, `drop` is also a chance to point users to other, more nuanced delete commands, and present a _big_ warning before they execute drop. These two together should help move users toward other commands that may match thier intent. Drop intentionally doesn't work for data  _authors_, forcing them to use `delete`, which clarifies they're making edits that will affect _everyone_ that relies on this dataset.
+
+Knowing `drop` exists should make data authors more comfortable with `push`. A safe feeling that a user can `push, drop, push, drop...` should help users feel safe knowing their actions can be undone. We _do_ need to pay careful attention to language here, letting users know that once something's pushed, we can't control who's _already pulled it_.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Use the logbook
+Delete actions should examine the history of a logbook to present warnings to a user that help them "clean up" when deleteing stuff. As an example: `$ qri delete --all` on a dataset should look for `push` operations and warn user they should `drop` from a list of remotes.
+
+### rename the `--drop` flag on `qri save`
+The current master branch has a `--drop` flag on the save command for "dropping components of a dataset". This RFC repurposes the `--drop` language to be about datasets you don't own, so this flag should be renamed to `--remove`. We shouldn't use `--delete`, because the word delete is now reserved for editing history.
+
+### Corner Cases
+There are a few scenarios we should make a point writing tests for:
+* Running amend on a dataset with no commits (no history) is an error
+* a drop request for a dataset name that doesn't exist locally, but _does_ exist on the remote should not error. This implies remotes should prefer local reference resolution
+* Remotes _should_ accept altered InitIDs when an owner Pushes to them. If user creates `b5/population`, pushes, then `delete --all b5/population` locally, then creates a new dataset with the same `b5/population` name, and pushes again, that should work, completely replacing the history
```

Comment:
FWIW, I feel like this is kind of a dangerous operation. For example, you wrote above:

> Knowing `drop` exists should make data authors more comfortable with `push`. A safe feeling that a user can `push, drop, push, drop...` should help users feel safe knowing their actions can be undone.

Obviously we’re talking about *“delete”* rather than “drop” here, but I feel like the essential thing above is that you’re suggesting someone should feel like something is safely backed up, or that there’s a safely/authoritatively stored copy somewhere else when they `push`. If they `push` an edited history, they’ve messed with the safe backup.

I think Git does actually solve this reasonably well by require a `--force` flag in this case. Qri could do the same. (Or potentially better by making the command interactive if in a TTY and `--force` is not specified?)

Reply:
> you’re suggesting someone should feel like something is safely backed up, or that there’s a safely/authoritatively stored copy somewhere else when they push. If they push an edited history, they’ve messed with the safe backup.

This *is* talking about `drop` (meaning un-push aka remove from remote), not `delete`.
The quote is supposed to convey the opposite: the user should feel comfortable `push`ing, because there is an easy way to remove it from the remote if they've made a mistake.

Reply:
> This is talking about drop (meaning un-push aka remove from remote), not delete.

I was more referring to the commands the user is using, but I get your point that `push` then `delete` then `push` is pretty similar to `drop --remote` (except `drop --remote` wouldn’t alter what you have locally, right?)

> The quote is supposed to convey the opposite: the user should feel comfortable `push`ing, because there is an easy way to remove it from the remote if they've made a mistake.

Right. I think I was more trying to say that I don’t think the user should feel comfortable here. For my part, I’ve certainly had more than one case where `git push` failing saved my butt because I didn’t intend for my local history and the remote’s histories to have diverged. This becomes an even bigger deal when multiple machines or people are pushing to the same dataset.